exporters/trace/jaeger/jaeger_test.go

```diff
@@ -886,3 +883,68 @@ func TestProcess(t *testing.T) {
 		})
 	}
 }
+
+type testCollectorEnpointWithBatches struct {
+	batchesUploaded *[]*gen.Batch
+}
+
+var _ batchUploader = (*testCollectorEnpointWithBatches)(nil)
+
+func (c *testCollectorEnpointWithBatches) upload(batch *gen.Batch) error {
+	*c.batchesUploaded = append(*c.batchesUploaded, batch)
+	return nil
+}
+
+func withTestCollectorEnpointWithSpans(batchesUploaded *[]*gen.Batch) func() (batchUploader, error) {
```

Comment:
It should be a pointer of the slice since slice `batchesUploaded` is expanded by `(*testCollectorEnpointWithBatches).upload` method to read later. 
But to simplify code, I am changing to reuse already exist struct - testCollectorEnpoint.